OpenRA.Mods.TS/Effects/AnimatedBeaconSprite.cs

```diff
@@ -0,0 +1,63 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using OpenRA.Effects;
+using OpenRA.Graphics;
+using OpenRA.Scripting;
+
+namespace OpenRA.Mods.TS.Effects
+{
+	public class AnimatedBeacon : IEffect
+	{
+		readonly Player owner;
+		readonly WPos position;
+		readonly string beaconPalette;
+		readonly bool isPlayerPalette;
+		readonly Animation beacon;
+
+		public AnimatedBeacon(Player owner, WPos position, int duration, string beaconPalette, bool isPlayerPalette, string beaconCollection, string beaconSprite)
+		{
+			this.owner = owner;
+			this.position = position;
+			this.beaconPalette = beaconPalette;
+			this.isPlayerPalette = isPlayerPalette;
+
+			if (!string.IsNullOrEmpty(beaconSprite))
+			{
+				beacon = new Animation(owner.World, beaconCollection);
+				beacon.PlayRepeating(beaconSprite);
+			}
+
+			if (duration > 0)
+				owner.World.Add(new DelayedAction(duration, () => owner.World.Remove(this)));
+		}
+
+		public void Tick(World world)
+		{
+			if (beacon != null)
+				beacon.Tick();
+		}
+
+		public IEnumerable<IRenderable> Render(WorldRenderer r)
+		{
+			if (!owner.IsAlliedWith(owner.World.RenderPlayer))
+				yield break;
+
+			if (beacon == null)
```

Comment:
Sorry to interrupt again, but I think we could move this check above the `IsAlliedWith` check to save some perf. ^^